src/sinks/aws_s3/config.rs

```diff
@@ -182,7 +184,14 @@ impl S3SinkConfig {
             .cloned()
             .unwrap_or_else(|| DEFAULT_KEY_PREFIX.into())
             .try_into()?;
-        let partitioner = KeyPartitioner::new(key_prefix);
+        let ssekms_key_id = self
+            .options
+            .ssekms_key_id
+            .as_ref()
+            .cloned()
+            .unwrap_or_else(|| DEFAULT_SSEKMS_KEY_ID.into())
+            .try_into()?;
+        let partitioner = S3KeyPartitioner::new(key_prefix, ssekms_key_id);
```

Comment:
I would prefer it if we passed the key id into the `S3KeyPartitioner` as an `Option<Template>` rather than use an empty string magic value. It's not critical, but it just makes the intent behind the code clearer.

Reply:
@StephenWakely I pushed up a commit with the changes I think you were looking for...can you take another look and see if I'm on the right track?